Update README (#516)
greenplum-db/pxf: README.md

make test
```

> ## How to Install PXF locally

Reviewer: I am not sure `locally` adds any value here, by default devs assume local install and all steps above were also for local install.

To install PXF, first make sure that the user has sufficient permissions in the `$GPHOME` and `$PXF_HOME` directories to perform the installation. It's recommended to change ownership to match the installing user. For example, when installing PXF as user `gpadmin` under `/usr/local/greenplum-db`:

```bash
export GPHOME=/usr/local/greenplum-db
export PXF_HOME=/usr/local/pxf

> export PXF_BASE=${HOME}/pxf-boot

Reviewer: ```suggestion
export PXF_BASE=${HOME}/pxf-base
```

chown -R gpadmin:gpadmin "${GPHOME}" "${PXF_HOME}"
make -C ~/workspace/pxf install
```

NOTE: if `PXF_BASE` is not set, it will default to `PXF_HOME`, and server configurations, libraries or other configurations, might get deleted after a PXF re-install.

## How to Run PXF locally

Ensure that PXF is in your path. This command can be added to your .bashrc
```bash
export PATH=/usr/local/pxf/bin:$PATH
```

Then you can prepare and start up PXF by doing the following.
```bash
pxf prepare
pxf start
```

> If `${HOME}/pxf-boot` does not exist, `pxf prepare` will create the directory for you. This command should only need to be run once.

Reviewer: ```suggestion
If `${HOME}/pxf-base` does not exist, `pxf prepare` will create the directory for you. This command should only need to be run once.
```

make test
```

## How to Install PXF locally

To install PXF, first make sure that the user has sufficient permissions in the `$GPHOME` and `$PXF_HOME` directories to perform the installation. It's recommended to change ownership to match the installing user. For example, when installing PXF as user `gpadmin` under `/usr/local/greenplum-db`:

> If `${HOME}/pxf-boot` does not exist, please create it: `mkdir ${HOME}/pxf-boot`. This command should only need to be run once.

Reviewer 2: actually, this is not necessary, you can run
```
PXF_BASE=${HOME}/pxf-boot pxf prepare
```
to `prepare` the `$PXF_BASE` directory. This includes creating the directory and copying the template files. This step has to be done after PXF has been installed

```bash
export GPHOME=/usr/local/greenplum-db
export PXF_HOME=/usr/local/pxf
export PXF_BASE=${HOME}/pxf-boot
chown -R gpadmin:gpadmin "${GPHOME}" "${PXF_HOME}"
make -C ~/workspace/pxf install
```

> NOTE: if `PXF_BASE` is not properly set up, there is a possibility that the user's existing server configurations may get deleted.

Reviewer 2: ```suggestion
NOTE: if `PXF_BASE` is not set, it will default to `PXF_HOME`, and server configurations, libraries or other configurations, might get deleted after a PXF re-install.
```

## How to Run PXF locally

Ensure that PXF is in your path. This command can be added to your .bashrc
```bash
export PATH=/usr/local/pxf/bin:$PATH
```

Then you can prepare and start up PXF by doing the following.
```bash

> pxf prepare

Reviewer 2: `pxf prepare` will create your PXF_BASE directory. We might want to note that this is a one-time step

pxf start
```

## Re-installing PXF after making changes

> Note: Local development with PXF requires a running GPDB6 cluster.

Reviewer 2: ```suggestion
Note: Local development with PXF requires a running Greenplum cluster.
```

Once the desired changes have been made, there are 2 options to re-install PXF:

1. Run `make -sj4 install` to re-install and run tests

> 2. Run `make -sj4 install-server` to only re-install PXF without running tests.

Reviewer 2: ```suggestion
2. Run `make -sj4 install-server` to only re-install the PXF server without running unit tests.
```

tar xzf singlecluster-HDP.tar.gz
```

Create a symlink using `ln -s ~/workspace/singlecluster-HDP ~/workspace/singlecluster` and then follow the steps in [Setup Hadoop](#Setup-Hadoop).

While PXF can run on either Java 8 or Java 11, please ensure that you are running Java 8 for hdfs, hadoop, etc. Please set your java version by seting your `JAVA_HOME` to the appropriate location.

On a Mac, you can set your java version using `JAVA_HOME` like so:
```
export JAVA_HOME=`/usr/libexec/java_home -v 1.8`
````

Initialize the default server configurations:
```

> cp ${PXF_HOME}/templates/{hdfs,mapred,yarn,core,hbase,hive}-site.xml ${PXF_BASE}/servers/default

Reviewer: ```suggestion
cp ${PXF_HOME}/templates/*-site.xml ${PXF_BASE}/servers/default
```
to also copy `pxf-site.xml`

tar xzf singlecluster-HDP.tar.gz
```

Create a symlink using `ln -s ~/workspace/singlecluster-HDP ~/workspace/singlecluster` and then follow the steps in [Setup Hadoop](#Setup-Hadoop).

Initialize the default server configurations:
```
cp ${PXF_HOME}/templates/{hdfs,mapred,yarn,core,hbase,hive}-site.xml ${PXF_BASE}/servers/default
```

# Development With Docker
NOTE: Since the docker container will house all Single cluster Hadoop, Greenplum and PXF, we recommend that you have at least 4 cpus and 6GB memory allocated to Docker. These settings are available under docker preferences.

> The following commands run the docker container and set up and switch to user gpadmin. They assume tat the user is running GPDB5

Reviewer 2: Not sure why we assume greenplum 5 is running, these instructions should work with all versions of Greenplum
```suggestion
The following commands run the docker container and set up and switch to user gpadmin.
```
